plugins/by-name/trouble/default.nix

```
@@ -300,14 +300,8 @@ helpers.neovim-plugin.mkNeovimPlugin {
     '';
   };
 
-  extraOptions = {
-    iconsPackage = lib.mkPackageOption pkgs [
-      "vimPlugins"
-      "nvim-web-devicons"
-    ] { nullable = true; };
-  };
-
   extraConfig = cfg: {
-    extraPlugins = mkIf (cfg.iconsPackage != null) [ cfg.iconsPackage ];
+    # TODO: added 2024-09-20 remove after 24.11
+    plugins.web-devicons.enable = mkOverride 1490 true;
```

I think we should be doing this only if `mini.icons` isn't enabled?

```suggestion
    plugins.web-devicons.enable = mkIf (!(config.plugins.mini.modules.icons.enable && config.plugins.mini.mockDevIcons)) (mkOverride 1490 true);
```

Having this littered throughout the codebase rather than defined in a central place isn't ideal, but I'm fine with it as this is probably a deprecation that we _actually_ plan to remove after 24.11.

Given some plugins have additional requirements for `plugins.web-devicons` to be enabled, I'm also fine with doing it separately in each plugin.

---

Hmm, originally I was just doing it to replicate the original behavior of our auto install in each of the plugins. But, the intention of the change is truly to support alternative icon providers so I agree it does make sense to change the auto install to only be done when a user isn't expecting the new icon provider to take effect.